nixos/modules/services/web-apps/mastodon.nix

```
@@ -175,6 +175,22 @@ let
       })
       (lib.range 1 cfg.streamingProcesses));
 
+  nginxCommonHeaders = ''
+    add_header Strict-Transport-Security 'max-age=31536000';
+  ''
+    # Upstream does not supported HTTP3 protocol.
+    + lib.optionalString (config.services.nginx.virtualHosts.${cfg.localDomain}.quic && config.services.nginx.virtualHosts.${cfg.localDomain}.http3) ''
+    add_header Alt-Svc 'h3=":$server_port"; ma=604800';
+  '';
+
+  nginxProxyHeaders = ''
+    proxy_set_header Host $host;
```

I've must have messed up something while testing and didn't really test fully. This line broke my setup because it contained the host of the instance twice seperated by `, `. This is probably because we have a haproxy with proxy procotol infront of it.

---

With proxy protocol I needed to set the following to fix the above error

```suggestion
    proxy_set_header Host $proxy_protocol_addr;
```
not idea how to solve this generic 😅

---

Add an additional parameter that only changes this value?

---

Actually there is no haproxy in front of the mastodon, it has it's own IP. Not sure where that is coming from.

---

Run the nginx server in debug mode?

---

Option `recommendedProxySettings` disabled?

---

> Run the nginx server in debug mode?

That will probably generate a ton of output I cannot parse /:


> Option recommendedProxySettings disabled?

It's on everywhere and here, too.

---

Can you publish the resulting nginx configuration? As I was told by Angie/nginx tech support, this situation can occur if the 'proxy_set_header Host $host;' parameter is used 2 times.

---

Have you checked the nginx configurations?